### PR TITLE
Fix Pack

### LINKED ISSUE #635: Magic Wall/ Wild Growth Duration
These runes should have a Random duration

Magic wall should be : between 16 ~ 20 seconds.

wg should be : between 38~ 45 seconds.

### LINKED ISSUE #793: NEW summons flag disappears
**Describe the bug**
I am using the most recent version of OTBR and the summons that only each vocation has have an error in the flag that indicates that it is a summon, when you go up or down the stairs the flag disappears, and will only reappear if you reconnect.

**To Reproduce**
1. Use the spell of your summon, of your vocation.
2. Go up some stairs or down, it also happens when you enter a teleport.

After this the flag disappeared.

 - OS: [Windows 10, Linux Debian 9]

### LINKED ISSUE #797: Market Imbuiment Item
Itens que já foram imbuidos anteriormente, mesmo com o imbuement removido, 
não são negociáveis no market.

### LINKED ISSUE #934: Summons Atravessando Paredes
Quando um monstro invocado por utevo res pisa em um field (imune ou não), ele atravessa paredes e objetos (de houses apenas) para encontrar uma forma de seguir seu mestre ou chegar ao seu alvo.

para reproduzir o bug basta:
- criar um druid ou sorcerer com level suficiente para utilizar utevo res
- ir em um ambiente fechado por uma porta ou paredes (exemplo simples: entre as casas Harbour Place 2 e Harbour Place 3, localizadas no norte do depot de Thais)
- invocar com um fire elemental (pode ser dentro ou fora do local)
- usar uma fire bomb rune sob os pés do summon
- faze-lo se mover
- tranca-lo (dentro ou fora do ambiente), e ficar no lado oposto da parede/porta

o summon irá buscar uma rota para acompanhar seu mestre, atravessando assim, qualquer objetivo, 
seja uma cama, parede, porta, janela, etc...

tive casos em que o summon até mesmo abriu a porta para passar (não consegui mais reproduzir o ocorrido

summonbug

após reproduzido o bug, o monstro continuará bugado e afetará sua outra summon, até ser morto

Reproduzido em um Windows Server 2012

### LINKED ISSUE #937: Pet not following
hi guys, when you summon your pet, and then you go to anothers level floors, the pet don't go with you, for example if i invoke my pet in floor 6 and i go to 7 floor the pet stay in level 6 xd

### PR DESCRIPTION
**Pull request by FakeShinoda**
**Imbuement**:

- _Fix #797_

**Summons behavior**:

- _Fix #793_
- _Fix #934_
- _Close #937_
- _Partially resolve issue #854_ **(need more information about the pets)**

**Misc**:

- _Fix #635_
- _Fixed issavi boat position_

Enjoy leechers, daddy blessings <3

### CODE REVIEW

File: data/spells/scripts/summon/summons.lua [hunk 1]
```diff
@@ -36,7 +36,7 @@ function onCastSpell(cid, var)
 
     if not summonName then return false end
 
-    local mySummon = Game.createMonster(summonName, player:getPosition(), true, false)
+    local mySummon = Game.createMonster(summonName, player:getPosition(), true, false, creature)
```
Comment (gerotib) [inline data/spells/scripts/summon/summons.lua]:
```suggestion
    local mySummon = Game.createMonster(summonName, player:getPosition(), true, false, cid)
```
either we have to change it to "cid" or whole function to `onCastSpell(creature, variant)`

**Comment by gerotib** [comment]
I tested summons, and with my suggested change both regular and pets getting correct flag while being summoned. So thats fixed, but pets are losing flag after relog, and then we need to change floor etc to "set" their flag like before.

File: data/spells/scripts/summon/summons.lua (hunk 1 above)
Comment (FakeShinoda) [reply data/spells/scripts/summon/summons.lua]: i'll change the whole function to creature/variant
Thanks

**Comment by FakeShinoda** [comment]
> 
> 
> I tested summons, and with my suggested change both regular and pets getting correct flag while being summoned. So thats fixed, but pets are losing flag after relog, and then we need to change floor etc to "set" their flag like before.

Master was not set in creaturescripts/pet.lua. Can you test again ?

**Comment by gerotib** [comment]
@FakeShinoda Yeah now it works. Summons also doesnt walk through house tiles, and walk around "walkable water". By the way this summon flag bug resisted in like every repo I checked, its awesome that finally someone managed to fix that, it was so annoying <3

**Comment by mattyx14** [comment]
check for example tenebris tile have special "invisible" if have this tile pet no appear

**Comment by FakeShinoda** [comment]
> 
> 
> check for example tenebris tile have special "invisible" if have this tile pet no appear

My question is, when the pet disappear the countdown stops? and when it ill be summoned again?

**Comment by mattyx14** [comment]
the time only pause and continue .. for example if monster dissaper 15 minutes y have time for summon the pet again .. send msg no possible

File: src/tile.cpp [hunk 2]
```diff
@@ -501,6 +501,14 @@ ReturnValue Tile::queryAdd(int32_t, const Thing& thing, uint32_t, uint32_t tileF
 				return RETURNVALUE_NOTPOSSIBLE;
 			}
 
+			if (monster->isSummon()) {
+				for (int i = ITEM_WALKABLE_SEA_START; i < ITEM_WALKABLE_SEA_END; i++) {
```
Comment (slawkens) [inline src/tile.cpp]: Shouldn't it be: i <= ITEM_WALKABLE_SEA_END ? As you want to include the last ID too or?

Comment (FakeShinoda) [reply src/tile.cpp]: > 
> 
> Shouldn't it be: i <= ITEM_WALKABLE_SEA_END ? As you want to include the last ID too or?

Fixed, with one less loop

File: data/spells/scripts/summon/summons.lua [hunk 3]
```diff
@@ -12,7 +12,7 @@ setCombatParam(combat, COMBAT_PARAM_EFFECT, CONST_ME_BLOCKHIT)
 local area = createCombatArea(AREA_CIRCLE1X1)
 setCombatArea(combat, area)
 
-function onCastSpell(cid, var)
+function onCastSpell(creature, variant)
```
Comment (gerotib) [inline data/spells/scripts/summon/summons.lua]:
```suggestion
function onCastSpell(player, variant)
```

File: data/spells/scripts/summon/summons.lua [hunk 4]
```diff
@@ -12,7 +12,7 @@ setCombatParam(combat, COMBAT_PARAM_EFFECT, CONST_ME_BLOCKHIT)
...
 	local player = Player(cid)
```
Comment (gerotib) [inline data/spells/scripts/summon/summons.lua]:
```suggestion

```

File: data/spells/scripts/summon/summons.lua [hunk 5]
```diff
@@ -36,9 +36,9 @@ function onCastSpell(cid, var)
 
     if not summonName then return false end
 
-    local mySummon = Game.createMonster(summonName, player:getPosition(), true, false)
+    local mySummon = Game.createMonster(summonName, player:getPosition(), true, false, creature)
```
Comment (gerotib) [inline data/spells/scripts/summon/summons.lua]:
```suggestion
    local mySummon = Game.createMonster(summonName, player:getPosition(), true, false, player)
```

**Review by gerotib — COMMENTED** [review]
Actually I messed up and didnt test it correctly last time, with player instead of creature this spell works fine
https://github.com/opentibiabr/otservbr-global/issues/937#issuecomment-603950844

**Comment by rtsrick2017** [comment]
@FakeShinoda is it possible to shorten the distance the summon teleports to the player? It takes too long for the summon to appear when you change floors, you have to take many steps, and this might be a problem in short distance caves with multiple floors

**Comment by daanyx** [comment]
Me parece que o Barco de (VENORE) Está levando para o tapete de Issavi.

**Comment by gerotib** [comment]
@rtsrick2017 what do you mean? For me summons (200+ ones) teleport immediately each time I change floor / walk 15+ tiles away. Only issue is that they start to follow after ~1 sec (probably some onThink trigger), so you can run away from them without getting followed in this time.

**Comment by majestyotbr** [comment]
Thanks for your contribution!

**Pull request merged**